celery/utils/dispatch/signal.py

```diff
@@ -181,6 +181,14 @@ def _connect_signal(self, receiver, sender, weak, dispatch_uid):
         if weak:
             ref = weakref.ref
             receiver_object = receiver
+            try:
```

Comment:
I think we should leave a comment here explaining why this code exists.

Reply:
Maybe we should even put the code into a utility function that deals with selecting the correct weakref type to be used for a Python object.

Reply:
That would be best.
I do think that the comment is required in any case.